### DISCUSSION TITLE
Toggle dark mode temporarily

### DISCUSSION THREAD

**Discussion by delanym** [community/community#137848] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Product Feedback

### Body

Generally I browse in dark mode. But occasionally I visit a page that has an image with a transparent background, and when the lines are dark, the image becomes unreadable. For example the sequence diagram at
https://github.com/darkweak/souin

The same could happen with white writing/lines against a light theme.

Please add button or something that will toggle the current theme with its opposite.

**Comment by feboyfierlyan** [comment]
Hi! A solution you might find useful is the Chrome extension called [Dark Reader](https://chromewebstore.google.com/detail/dark-reader/eimadpbcbfnmbkopoojfekhnkhdbieeh). It automatically adjusts website themes and even provides a toggle to adjust the brightness and contrast, making images with transparent backgrounds more readable. I hope this helps!
![image](https://github.com/user-attachments/assets/e6e04a1a-6615-471a-9dc7-2e9b1287c2cf)

**Reply by feboyfierlyan** [reply]
If you find this post helpful, don't forget to accept my answer! <3

**Reply by delanym** [reply]
Thanks but I don't use Chrome

**Comment by AVMiraga** [comment]
If you are asking Temporary "Temporary", this JS script will do it:
```javascript
javascript:(function() {
  var html = document.documentElement; 
  var currentMode = html.getAttribute('data-color-mode'); 
  if (currentMode === 'light') {
    html.setAttribute('data-color-mode', 'dark');
  } else {
    html.setAttribute('data-color-mode', 'light');
  }
})();
``` 

Assuming you have a bookmark logic in your browser and it supports JS running.

Bookmark this js script and whenever you click it, it will change mode.

========

Alternatively, @feboyfierlyan mentioned darkreader, which is supported by various browsers, not only Chrome. Sometimes that extension behaves in an unintended way, where inverts the image color too.

**Reply by AVMiraga** [reply]
As Temporary "Temporary", I mean it will only be applied to one page, you need to inject (run/execute) that JS script again on another page. Another approach is cookie logic which is not working because of gh_session, probably it is keeping theming. color_mode cookie keeps that JSON data which could be manipulated, but as I said it gets overwritten by something else.

**Comment by matheusqmello** [comment]
If it makes sense for you, you can edit 1 attribute of "html" tag on github website, using the browser developer tool/inspect sourcecode:
- `data-color-mode="dark"` --> change to `"light"` to see git in light theme and vice-versa;

Example:
`<html lang="en" data-color-mode="light" data-light-theme="light" data-dark-theme="dark" data-a11y-animated-images="system" data-a11y-link-underlines="true" data-turbo-loaded="">`
